contracts/interchain-token/InterchainToken.sol

```diff
@@ -15,10 +16,15 @@ import { Minter } from '../utils/Minter.sol';
  * @title InterchainToken
  * @notice This contract implements an interchain token which extends InterchainToken functionality.
  * @dev This contract also inherits Minter and Implementation logic.
+ * Uses assembly to directly write to slot 0, bypassing parent contract storage layout.
  */
 contract InterchainToken is InterchainTokenStandard, ERC20, ERC20Permit, Minter, IInterchainToken {
     using AddressBytes for bytes;
 
+    /// @dev Storage slot for deployer address - FORCED to slot 0 using assembly
+    /// This is specifically slot 0 for Hyperliquid firstStorageSlot compatibility
+    bytes32 private constant DEPLOYER_SLOT = 0x0000000000000000000000000000000000000000000000000000000000000000;
```

Comment:
The slot 0 needs to be reserved explicitly. Need to add a variable for it at the top.

Foivos recommended creating a separate HyperLiquidDeployer contract and inheriting from it first to do this cleanly